schemas/user.yml

```diff
@@ -81,3 +81,20 @@
           Name of the directory the user is a member of.
 
           For example, an LDAP or Active Directory domain name.
+
+    - name: roles
```

Comment:
@ebeahan Over on #915 you were floating the idea of other names for the field. I think the simpler `.roles` name would work great specifically because it's not qualifying the field name too much. This way the field may work for more situations (wild guess).

What kind of ambiguities do you think the other suggested names resolve?

Reply:
I'm going to pull an IAM example from the proposed `user.target` field since I think it helps capture my thinking.

The `user.roles` field values are being populated with two purposes: 

1. the `elastic` user is assigned the `superuser` role. Here `.roles` captures `elastic's` assigned roles at the time of the event.
2. `elastic` is modifying user `bob` and changing `bob's` roles. Roles here captures the changes to bob's user permissions being assigned in the system.

```json
{
  "user": {
    "name": "elastic",
    "roles": [ "superuser" ]
    "target": {
      "name": "bob"
    },
    "changes": {
      "roles": [ "kibana_user", "superuser" ]
    }
  }
}
```

It's only a minor difference though - what roles are currently assumed by a user vs. the assigning of roles to a user.

Reply:
So `user.target` should also contain the prior state for the roles. If it had none, it should have an empty array. If it had e.g. "kibana_user" and is being added "superuser", it should look like this:

```JSON
{
  "user": {
    "name": "elastic",
    "roles": [ "superuser" ],
    "target": {
      "name": "bob",
      "roles": [ "kibana_user" ]
    },
    "changes": {
      "roles": [ "kibana_user", "superuser" ]
    }
  }
}
```

Reply:
Ok. Do you think we should limit (in the field's description) the scope of `user.roles` to just be the assigned roles/privileges of the user performing the event? Or is it ok if in implementation the field might get used to track assignment of those roles too?

Reply:
@webmat made a good point that using categorization fields could help the user filter between the different use cases I had some minor concerns. 👍